File: src/main/Java/boj_step14/boj1764.java
import java.io.*;
import java.util.*;

public class boj1764 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        HashSet<String> noListen = new HashSet<>();
        HashSet<String> noLook = new HashSet<>();
        List<String> noListenLook = new ArrayList<>();
        while (n-- > 0) noListen.add(reader.readLine());
        while (m-- > 0) noLook.add(reader.readLine());
        for (String noListner: noListen) if (noLook.contains(noListner)) noListenLook.add(noListner);
        Collections.sort(noListenLook);
        writer.append(noListenLook.size() + "\n");
        for (String it: noListenLook) writer.append(it + "\n");
        writer.close();
    }
}

File: src/main/boj_step14/boj1764.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, m) = r.readLine().split(" ").map { it.toInt() }
    val noListen = mutableSetOf<String>()
    val noLook = mutableSetOf<String>()
    var set = mutableSetOf<String>()
    repeat(n) {
        noListen.add(r.readLine())
    }
    repeat(m) {
        noLook.add(r.readLine())
    }
    noListen.forEach {
        if (noLook.contains(it)) {
            set.add(it)
        }
    }
    w.write(set.size.toString())
    w.newLine()
    set = set.sorted().toMutableSet()
    set.forEach {
        w.write(it)
        w.newLine()
    }
    w.close()
}
